maven/new_rules.bzl

```diff
@@ -0,0 +1,268 @@
+load("@graknlabs_bazel_distribution//maven:rules.bzl", "MavenDeploymentInfo")
+
+def _parse_maven_coordinates(coordinates_string):
+    group_id, artifact_id, version = coordinates_string.split(":")
+    if version != "{pom_version}":
+        fail("should assign {pom_version} as Maven version via `tags` attribute")
+    return struct(
+        group_id = group_id,
+        artifact_id = artifact_id,
+    )
+
+def _parse_maven_artifact(coordinates_string):
```

Comment:
should this method be called `_parse_maven_artifact`? it is also very similar to `_parse_maven_coordinates`. can't we combine these two into one generic function?